src/index.js

```diff
@@ -69,76 +60,151 @@ class Saturn {
       clearTimeout(connectTimeout)
 
       const { headers } = res
-      log.ttfb = new Date()
+      log.ttfbMs = new Date() - log.startTime
       log.httpStatusCode = res.status
-      log.cacheStatus = headers.get('saturn-cache-status')
-      log.nodeId = headers.get('saturn-node-id') ?? headers.get('x-ipfs-pop')
+      log.cacheHit = headers.get('saturn-cache-status') === 'HIT'
+      log.nodeId = headers.get('saturn-node-id')
       log.transferId = headers.get('saturn-transfer-id')
       log.httpProtocol = headers.get('quic-status')
 
       if (!res.ok) {
-        throw new Error(`Non OK response received: ${res.status} ${res.statusText}`)
+        throw new Error(
+          `Non OK response received: ${res.status} ${res.statusText}`
+        )
       }
+    } catch (err) {
+      log.ifNetworkError = err.message
+      // Report now if error, otherwise report after download is done.
+      this._finalizeLog(log)
+
+      throw err
+    } finally {
+      this._addPerformanceAPIMetrics(log)
+    }
 
-      const validationResult = await (options.downloadTimeout
-        ? Promise.race([validateBody(res.body), setTimeoutPromise(options.downloadTimeout, false, { ref: false })])
-        : validateBody(res.body))
+    return { res, log }
```

Comment:
I wanted to keep the simpler and focused `return res`, but this was the simplest way to pass a log alongside a response.

Could've also done this.. but idk 🤔 .  thoughts?
```js
res.log = log

return res
```

Reply:
I prefer passing both in an object. `res.log` doesn't really click with me

Reply:
Better explicit than magic, I also prefer returning both